app/views/shared/transactions/_search2.html.haml

```diff
@@ -1,6 +1,6 @@
 -# This view, replaces _search, which relies on the TransactionSearch controller/concern.
 -# This view, along with TransactionSearch::Searcher is the more modern method. See FacilityAccountsReconciliationController
-= simple_form_for @search_form, url: url_for(params), method: :get, html: { class: "search_form" }, defaults: { required: false } do |f|
+= simple_form_for @search_form, url: url_for(params.permit(:search)), method: :get, html: { class: "search_form" }, defaults: { required: false } do |f|
```

Comment:
I'm surprised this works without additional `permit(:product_id, ...)`, but hey, if it works, awesome.